test/runtests.jl

```diff
@@ -57,18 +57,54 @@ end
 
 const TEST_PKG = (name = "Example", uuid = UUID("7876af07-990d-54b4-ab0e-23690620f79a"))
 
-@testset "Runner tests" begin
-    # Run tests for an installed package
-    Pkg.add(TEST_PKG.name)
-    TestReports.test("Example") # Should pass
-    Pkg.rm(TEST_PKG.name)
-    @test_throws Exception TestReports.test("Example")
-
-    # Run tests for a local, in-development package
-    Pkg.develop(TEST_PKG.name; shared = false)
-    Pkg.activate(TEST_PKG.name)
-    TestReports.test(TEST_PKG.name)
+"""
+Helper function to verify the runner successfully ran a test suite based on
+the Info messages that were logged.
+"""
+function test_successful_testrun(testrun::Function, project::AbstractString)
+    successful_pass_matcher = Regex("$(project) tests passed. Results saved to .+\\.\$")
+    @test_logs (:info, "Testing $(project)") (:info, successful_pass_matcher) testrun()
+end
+
+""""
+Helper function to execute 'work' within the context of an active
+project. The `project` should be `Pkg.develop`able.
+"""
+function with_active_project(work_within_context::Function, project::AbstractString)
+    Pkg.develop(project; shared = false)
+    Pkg.activate(project)
+
+    test_successful_testrun(work_within_context, project)
+
     Pkg.activate()
-    Pkg.rm(TEST_PKG.name)
-    @test_throws Exception TestReports.test(TEST_PKG.name)
+    Pkg.rm(project)
+end
+
+@testset "Runner tests" begin
+    @testset "installed packages by name" begin
+        Pkg.add(TEST_PKG.name)
+        test_successful_testrun(() -> TestReports.test(TEST_PKG.name), TEST_PKG.name)
+        Pkg.rm(TEST_PKG.name)
+        @test_throws TestReports.PkgTestError TestReports.test("Example")
+    end
+
+    @testset "activated projects" begin
+        @testset "by name" begin
+            # The test run should not fail when passed the name of the project
+            # that is activated and fail when its deactivated
+            with_active_project(TEST_PKG.name) do
+                TestReports.test(TEST_PKG.name)
+            end
+            @test_throws TestReports.PkgTestError TestReports.test(TEST_PKG.name)
+        end
+
+        @testset "implicitly without positional arguments" begin
+            # The test run should not fail when a project is implied through
+            # one being active and fail with a descriptive error otherwise
+            with_active_project(TEST_PKG.name) do
+                TestReports.test()
+            end
+            @test_throws TestReports.PkgTestError TestReports.test()
```

Comment:
It looks like [#15.2](https://travis-ci.com/github/JuliaTesting/TestReports.jl/jobs/363647241) fails as this test is assumed to be run in the context where no project is active. However, on Travis deactivating the `Example` package/project reactivates the `TestReports` package/project which causes the job to go into a loop and fail.

Reply:
This is caused by the presence of the [`JULIA_PROJECT` environment variable](https://travis-ci.com/github/JuliaTesting/TestReports.jl/jobs/363647241#L185), which makes a plain `Pkg.activate()` revert back to that specific project. I've been able to reproduce the behavior locally.

The easiest 'solution' is to drop the test for 'no active project' case, but I don't think that is the right thing to do here. I'll investigate further.

Reply:
Resolved in 04fc4d276b740f5ed3fcbe7118cc1fa6c0e80ad6.